asset/js/setup.js

```diff
@@ -19,13 +19,48 @@ function flattenModals() {
   });
 }
 
+function insertCss(cssCode) {
+  const newNode = document.createElement('style');
+  newNode.innerHTML = cssCode;
+  document.getElementsByTagName('head')[0].appendChild(newNode);
+}
+
 function setupAnchors() {
+  const navBarSelector = jQuery('nav, .navbar');
+  const isFixed = navBarSelector.attr('style').includes('position: fixed;');
+  const navbarHeight = navBarSelector.height();
+  if (isFixed) {
+    jQuery('.nav-inner').css('padding-top', `calc(${navbarHeight}px + 1rem)`);
+  }
   jQuery('h1, h2, h3, h4, h5, h6, .header-wrapper').each((index, heading) => {
     if (heading.id) {
       jQuery(heading).on('mouseenter',
                          () => jQuery(heading).find('.fa.fa-anchor').css('visibility', 'visible'));
       jQuery(heading).on('mouseleave',
                          () => jQuery(heading).find('.fa.fa-anchor').css('visibility', 'hidden'));
+      if (isFixed) {
+        /**
+         * Fixing the top navbar would break anchor navigation,
+         * by creating empty spans above the <h> tag we can prevent
+         * the headings from being covered by the navbar.
+         */
+        const spanId = heading.id;
+        jQuery(heading).append(
+          jQuery('<span/>',
+                 {
+                   id: spanId,
+                   class: 'anchor',
+                 }));
+        jQuery(heading).removeAttr('id'); // to avoid duplicated id problem
+        const headingHeight = jQuery(heading).height();
+        const heightOffset = navbarHeight + headingHeight;
+        const spanCss = `span#${spanId} { margin-top: calc(-${heightOffset}px - .5rem);\n`
```

Comment:
Just wondering - why do we need to use the heading height in the calculation? Docusaurus seems to just use a static value. Would just using the height of the navbar + some clearance suffice for us? This would allow us to just use a static style instead of adding CSS dynamically like this.

I'd love to fiddle around with this a bit more to see what works, but I think the undefined `attr('style')` mentioned above seems to break the netlify demo. 😛 Let me know if you want me to take another look at this after you have fixed the issue.

Reply:
@damithc mentioned a case where he added some extra stuff in the header: https://nus-cs2103-ay1920s1.github.io/website/index.html
In this case we have to dynamically calculate the height.

(I used a fixed number in the previous PR...

Reply:
Hmm, I'm still not sure how the heading height takes into account additional things in the header. Do you have a test site with this that I can look at?

I've fiddled around with this a bit and I think this is a way we can do it without having to dynamically compute the `headingHeight` for each individual heading.
1. Move the dummy anchor span above the heading (instead of below it)
2. Set `display: block` and `position: relative` - we want to "float" this span above the actual heading, so that when this span is at the top of the page, our heading will be under the navbar
3. Set `top` to be negative the height of the `navbar`

This is what the outcome would look like:
<img width="807" alt="Screenshot 2020-01-20 at 5 48 53 PM" src="https://user-images.githubusercontent.com/17447681/72716582-6ea66100-3bad-11ea-8488-a23133ae3732.png">

This way, we can just use a common class for all anchors, instead of having to define a style separately for each anchor.